tests/pds/dataservice_test.go

```diff
@@ -130,12 +129,182 @@ var _ = BeforeSuite(func() {
 	Step("Create/Get Namespace and NamespaceID", func() {
 		namespace = pdslib.GetAndExpectStringEnvVar(envNamespace)
 		Expect(namespace).NotTo(BeEmpty(), "ENV "+envNamespace+" is not set")
+		isavailabbe, err := pdslib.CheckNamespace(namespace)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(isavailabbe).To(BeTrue())
 		namespaceID, err = pdslib.GetnameSpaceID(namespace, deploymentTargetID)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(namespaceID).NotTo(BeEmpty())
 	})
 })
 
+var _ = Describe("{ScaleUPDataServices}", func() {
+
+	JustBeforeEach(func() {
+		if !DeployAllDataService {
+			supportedDataServices = append(supportedDataServices, DataService)
+			for _, ds := range supportedDataServices {
+				logrus.Infof("supported dataservices %v", ds)
+			}
+			Step("Get the resource and app config template for supported dataservice", func() {
+				dataServiceDefaultResourceTemplateIDMap, dataServiceNameIDMap, err = pdslib.GetResourceTemplate(tenantID, supportedDataServices)
+				Expect(err).NotTo(HaveOccurred())
+
+				dataServiceNameDefaultAppConfigMap, err = pdslib.GetAppConfTemplate(tenantID, dataServiceNameIDMap)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(dataServiceNameDefaultAppConfigMap).NotTo(BeEmpty())
+			})
+			oldDSImage = pdslib.GetAndExpectStringEnvVar(ImageToBeUpdated)
+			oldDSVersion = pdslib.GetAndExpectStringEnvVar(VersionToBeUpdated)
+		}
+		updatedRep, err := pdslib.GetAndExpectIntEnvVar(envUpdateReplicas)
+		Expect(err).NotTo(HaveOccurred())
+		updatedrep = int32(updatedRep)
+
+	})
+
+	It("scale up dataservice nodes", func() {
+		Step("Deploy Data Services", func() {
```

Comment:
Are we deploying dataservice even when DeployAllDataService is false?

Reply:
Yes, we are deploying data service if DeployAllDataService is set to false.
we have two env params  DeployAllDataService and DataService.
If DeployAllDataService flag is required to run `DeployAllDataServices` test, For other test we need only DataService flag to be set